newrelic/resource_newrelic_synthetics_private_location.go

```diff
@@ -0,0 +1,193 @@
+			},
+			"key": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The private locations key.",
+			},
+			"location_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "An alternate identifier based on name.",
+			},
+		},
+	}
+}
+
+func resourceNewRelicSyntheticsPrivateLocationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+	var diags diag.Diagnostics
+
+	description := d.Get("description").(string)
+	name := d.Get("name").(string)
+	verifiedScriptExecution := d.Get("verified_script_execution").(bool)
+	res, err := client.Synthetics.SyntheticsCreatePrivateLocationWithContext(ctx, accountID, description, name, verifiedScriptExecution)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(res.Errors) > 0 {
+		for _, err := range res.Errors {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  err.Description,
+			})
+		}
+	}
+
+	if len(diags) > 0 {
+		return diags
+	}
+	d.SetId(string(res.GUID))
+	_ = d.Set("domain_id", res.DomainId)
+	_ = d.Set("key", res.Key)
+	_ = d.Set("location_id", res.LocationId)
+	_ = d.Set("guid", res.GUID)
+
+	return nil
+}
+
+func resourceNewRelicSyntheticsPrivateLocationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfig).NewClient
+	log.Printf("[INFO] Reading New Relic Synthetics Private Location %s", d.Id())
+
+	guid := common.EntityGUID(d.Id())
+	resp, err := client.Entities.GetEntity(guid)
+	if err != nil {
+		if _, ok := err.(*errors.NotFound); ok {
+			d.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	setCommonSyntheticsPrivateLocationAttributes(resp, d)
+	return nil
+}
+
+func setCommonSyntheticsPrivateLocationAttributes(v *entities.EntityInterface, d *schema.ResourceData) {
+	switch e := (*v).(type) {
+	case *entities.GenericEntityOutline:
+		_ = d.Set("guid", e.GUID)
+		_ = d.Set("name", e.Name)
+
+	}
+}
+
+func resourceNewRelicSyntheticsPrivateLocationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	var diags diag.Diagnostics
+	description := d.Get("description").(string)
+	guid := synthetics.EntityGUID(d.Id())
+	verifiedScriptExecution := d.Get("verified_script_execution").(bool)
+	res, err := client.Synthetics.SyntheticsUpdatePrivateLocation(description, guid, verifiedScriptExecution)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(res.Errors) > 0 {
+		for _, err := range res.Errors {
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  err.Description,
+			})
+		}
+	}
+
+	if len(diags) > 0 {
+		return diags
+	}
+
```

Comment:
Does the update return the same fields as create? https://github.com/newrelic/terraform-provider-newrelic/pull/1826/files#diff-17a90def9acc51921f5dd27174a27592bd4f3b84c081e22266b41c16d8d2567bR102

If so we should probably update them, in case any of them change after an update.